NGitLab.Tests/Impl/JsonConverterTests.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using NGitLab.Impl.Json;
+using NUnit.Framework;
+
+namespace NGitLab.Tests.Impl
+{
+    public class JsonConverterTests
+    {
+        [Test]
+        public void Test_DeserializeNullToSupportedValueType_Succeeds()
+        {
+            var json = @"{
+  ""a_boolean"": null,
+  ""a_date_time"": null,
+  ""a_double"": null,
+  ""an_int32"": null,
+  ""an_int64"": null
+}";
+            var obj = Serializer.Deserialize<MyDataContract>(json);
+
+            Assert.NotNull(obj);
+            Assert.AreEqual(false, obj.SomeBool);
+            Assert.AreEqual(DateTime.MinValue, obj.SomeDateTime);
+            Assert.AreEqual(0.0, obj.SomeDouble);
+            Assert.AreEqual(0, obj.SomeInt32);
+            Assert.AreEqual(0L, obj.SomeInt64);
+        }
+
+        [Test]
+        public void Test_DeserializeNullToUnsupportedValueType_Throws()
+        {
+            var json = @"{ ""a_uint32"": null }";
+            var ex = Assert.Throws<JsonException>(() => Serializer.Deserialize<MyDataContract>(json));
+            StringAssert.StartsWith("The JSON value could not be converted to System.UInt32.", ex.Message);
+        }
+
+        [Test]
+        public void Test_DeserializeStringToInt32_Throws()
+        {
+            var json = @"{ ""an_int32"": ""1234"" }";
+            var ex = Assert.Throws<JsonException>(() => Serializer.Deserialize<MyDataContract>(json));
+            StringAssert.StartsWith("The JSON value could not be converted to System.Int32.", ex.Message);
+        }
+
+        [Test]
+        public void Test_DeserializeStringToInt64_Succeeds()
+        {
+            var json = @"{ ""an_int64"": ""1234"" }";
+            var obj = Serializer.Deserialize<MyDataContract>(json);
+            Assert.NotNull(obj);
+            Assert.AreEqual(1234L, obj.SomeInt64);
+        }
+
+        [TestCase("2022-01-12")]
+        [TestCase("2022-01-12T22:49:21.552Z")]
+        [TestCase("2022-01-12T22:49:21.552+00:00")]
+        public void Test_DeserializeStringToDateTime_SupportsMultipleFormats(string input)
+        {
+            var json = $@"{{ ""a_date_time"": ""{input}"" }}";
+            var obj = Serializer.Deserialize<MyDataContract>(json);
+            Assert.NotNull(obj);
+            Assert.AreEqual(DateTimeKind.Utc, obj.SomeDateTime.Kind);
```

Comment:
Should it be unknown for `[TestCase("2022-01-12")]`?

Reply:
Yeah, I guess that would make sense. I'll revisit the code...

Reply:
Done.

Also, I removed unnecessary `DateTime` format strings, i.e. those without which the tests still pass.